cou/apps/app.py

```diff
@@ -91,31 +100,31 @@ def create(
 
     @classmethod
     def register_application(
-        cls, app_types: list[str]
+        cls, families: list[str]
     ) -> Callable[[type[OpenStackApplication]], type[OpenStackApplication]]:
         """Register Application subclasses.
 
         Use this method as decorator to register Applications that
         cannot be described appropriately by the OpenStackApplication class.
 
-        Example:
-        ceph_types = ["ceph-mon", "ceph-fs", "ceph-radosgw", "ceph-osd"]
-
-        @AppFactory.register_application(ceph_types)
-        class Ceph(OpenStackApplication):
+        @AppFactory.register_application(["ceph", "rabbitmq-server"])
```

Comment:
The benefit it's not populating charms from a family like we were doing on  this [comment](https://github.com/canonical/charmed-openstack-upgrader/pull/42#discussion_r1317257747). Now the application knows witch family they came from and this helps the auxiliary charms find their right channel track

Reply:
when we had that discussion, I did not understand what `add_charms` was for. I have now checked it out and iiuc you're fundamentally trying to manage in a reasonable way a 3-dimensional data structure with aliases: on one hand you have to map series, o7k codenames, and charm names to a track, on the other you have several duplicates so you want to have aliases.
Additionally, you want the mappings to be easy to update when there's a new release.

I think it may be valuable to reframe this as a data structure/storage problem rather than a method/decorator problem, as solving this from the Python logic side makes things less readable.

I would suggest exploring using csv tables (like you did for workload mappings) as they are easy to update so you may not care anymore about adding support for charm name aliases. csvs are only 2D of course, so you'd have to choose how to step one dimension down, for example by having per-LTS tables. You could still then reconstruct a sort of 3D structure by loading them into a single multilevel dictionary (to be queried as `mapping[series][codename][charm]` or any other level permutation.

Another alternative would be using pandas mulitindexes, but I don't think it's worth it for our usecase.

Reply:
Based on our conversation from yesterday, it was decided to create a single csv file with all charms and not need to populate by charm families.